lmod: depends on pkgconf

SeanChang-3245 opened the pull request:
<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->
Lmod's configure script explicitly looks for `pkg-config` executable , so I added an explicit build-time dependency on `pkgconf` to ensure the configuration phase succeeds.

Fixes: spack/spack#50348

SeanChang-3245 committed:
lmod: depends on pkgconf

SeanChang-3245 committed:
Merge branch 'develop' into develop

alalazo commented on repos/spack_repo/builtin/packages/lmod/package.py:
```diff
@@ -66,6 +66,8 @@ class Lmod(AutotoolsPackage):
 
     depends_on("bc", type="build", when="@8.7.10:")
 
+    depends_on("pkgconf", type="build")
```
```suggestion
    depends_on("pkgconfig", type="build")
```

This is because:
```
$ spack providers pkgconfig
pkgconfig:
-- no arch / no compilers ---------------------------------------
pkg-config  pkgconf
```

SeanChang-3245 replied:
Got it, thanks!

SeanChang-3245 committed:
lmod: depends on pkgconfig

Depends on virtual package `pkgconfig` instead of a concrete package like `pkgconf`

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>
